mirgecom/steppers.py

```diff
@@ -80,12 +81,13 @@ def _advance_state_stepper_func(rhs, timestepper, checkpoint, get_timestep,
         if logmgr:
             logmgr.tick_before()
 
-        dt = get_timestep(state=state)
-        if dt < 0:
-            return istep, t, state
+        if get_timestep:
+            dt = get_timestep(state=state)
+            if dt < 0:
+                return istep, t, state
 
         if checkpoint:
-            checkpoint(state=state, step=istep, t=t, dt=dt)
+            state, dt = checkpoint(state=state, step=istep, t=t, dt=dt)
```

Comment:
```suggestion
            dt = checkpoint(state=state, step=istep, t=t, dt=dt)
```

Reply:
Fixed this together with the other places it needed fixin.